### ISSUE TITLE
[bug] False positives when using `@eslint-react/hooks-extra/prefer-use-state-lazy-initialization`

### ISSUE THREAD

**Issue by dd-jonas** [Rel1cx/eslint-react#711]
### Describe the bug

Hi, I am using the rule `@eslint-react/hooks-extra/prefer-use-state-lazy-initialization` but noticed that using lazy initialization doesn't fix the lint error. I.e. changing `useState(getValue())` to `useState(() => getValue())` still gets reported.

I looked around in the code and noticed there are no actual tests that assert `useState(() => getValue())` should pass this rule. I'm not familiar enough with AST's to see where the error in the code is though.

### Reproduction

https://stackblitz.com/edit/vitejs-vite-y7xgx2?file=src%2FApp.tsx

Run `npm run lint` in the terminal to see the error.

### Expected behavior

No linting error because the useState call uses lazy initialization.

### Platform and versions

```Text
Node v20
@eslint-react/eslint-plugin v1.10.0
```

**Comment by Rel1cx** [comment]
Thank you for providing the reproduction. It seems that the modification to `getNestedCallExpressions` in [this commit](https://github.com/Rel1cx/eslint-react/commit/512389a6d7f55c06621d49a80dd88046e9a4ec07#diff-793ce44fe58268631679385871a33cac1a5ea1302f562edf5d69a0632ef643b6R9) has altered the semantics of this method, leading to this issue.

**Comment by dd-jonas** [comment]
@Rel1cx Thanks for addressing this so quickly!
